server/website/website/tasks/async_tasks.py

```diff
@@ -1172,11 +1091,11 @@ def map_workload(map_workload_input):
             'rowlabels': rowlabels,
         }
 
-    if len(workload_data) == 0:
+    if len(workload_data) < 2:
```

Comment:
I don't think this is correct - if there are 2+ workloads then the workload data for the current session will still be added to `workload_data` and this check does not catch that. You need to explicitly check if the unique workload is the workload for this session and skip processing it if it is.

Reply:
This is just a first-step filter. I check the workload afterward. Please see the below comment.

Reply:
If there is more than 1 workload, it is guaranteed that there is a mapped workload, so we do not need to return early here.